lib/AST/AutoDiff.cpp

```diff
@@ -0,0 +1,68 @@
+//===--------- AutoDiff.cpp - Swift Differentiable Programming ------------===//
```

Comment:
nit: please use the standard header guard, which left aligns the filename, and please change the thing after it to be descriptive, not a marketing thing :-)

Reply:
Done in https://github.com/apple/swift/pull/28738/commits/2ce6b05d6efcbd6a1c9a1c95d2bd6597fac57252.

I noticed that `IndexSubset.{h,cpp}` also use inconsistent headers, so I fixed those in https://github.com/apple/swift/pull/28757.